src/internal/packager/helm/chart.go

```diff
@@ -28,74 +31,87 @@ import (
 	"k8s.io/apimachinery/pkg/runtime"
 	"sigs.k8s.io/yaml"
 
-	"github.com/zarf-dev/zarf/src/config"
 	"github.com/zarf-dev/zarf/src/internal/healthchecks"
+	"github.com/zarf-dev/zarf/src/internal/packager/template"
 	"github.com/zarf-dev/zarf/src/pkg/message"
 	"github.com/zarf-dev/zarf/src/types"
 )
 
 // Use same default as Helm CLI does.
 const maxHelmHistory = 10
 
+type InstallUpgradeOpts struct {
```

Comment:
opts ++

I'm curious if we can simplify this further by snapshotting both `cluster`and `state`and passing in immutable copies rather than the entire world.

Reply:
Also, I'm typically in favor of commenting public fields on options structs. Usually I try to explain the capture the intended usage for each field. It can get repetitive though - kind of depends on the style guide for the team. Thoughts?

Reply:
I'm 50/50. I'm also okay with not commenting every field, but leaning towards comments. Cluster for example, feels pretty obvious in the context, it's a struct that can really only be used to connect to a k8s cluster. I like the way you framed capturing the intended use of the fields. I added comments with that in mind.

Reply:
Re: immutable data: This was intended as an architectural consideration for the future, no need to balloon scope on this PR to address is now. I'll make that more clear in my comments next time 😄

Re: doccomments: Nice, yeah that strikes a good balance and aligns well with the guidance in the `Types` section here: https://tip.golang.org/doc/comment#Types